lib/openldap.c

```diff
@@ -205,8 +259,68 @@ static CURLcode oldap_setup_connection(struct Curl_easy *data,
   return CURLE_OK;
 }
 
+/* Starts LDAP simple bind. */
+static CURLcode oldap_perform_bind(struct Curl_easy *data, ldapstate newstate)
+{
+  CURLcode result = CURLE_OK;
+  struct connectdata *conn = data->conn;
+  struct ldapconninfo *li = conn->proto.ldapc;
+  char *binddn = NULL;
+  struct berval passwd;
+  int rc;
+
+  passwd.bv_val = NULL;
+  passwd.bv_len = 0;
+
+  if(conn->bits.user_passwd) {
+    binddn = conn->user;
+    passwd.bv_val = conn->passwd;
+    passwd.bv_len = strlen(passwd.bv_val);
+  }
+
+  rc = ldap_sasl_bind(li->ld, binddn, LDAP_SASL_SIMPLE, &passwd,
+                      NULL, NULL, &li->msgid);
+  if(rc == LDAP_SUCCESS)
+    state(data, newstate);
+  else
+    result = oldap_map_error(rc, conn->bits.user_passwd?
+                                 CURLE_LOGIN_DENIED: CURLE_LDAP_CANNOT_BIND);
+  return result;
+}
+
 #ifdef USE_SSL
 static Sockbuf_IO ldapsb_tls;
```

Comment:
I know you didn't add this, but this is really bad isn't it? A static variable like this seems like a potential breaker of many things if we do many LDAP parallel transfers, right?

Reply:
Strike that. This struct is just full of function pointers and should be fine to reuse.

Reply:
I don't like this very much either. But as you say, it should'nt harm.